prow/plugins/plugins.go

```diff
@@ -446,3 +448,12 @@ func NewMetrics() *Metrics {
 		ConfigMapGauges: configMapSizeGauges,
 	}
 }
+
+func ReadSnippet(plugin string) string {
```

Comment:
This is  a cool idea but I think we should be able to leverage the same source as `godoc` to autogenerate these from the types we've defined, otherwise we risk this continually being out of sync.

Reply:
Ok, so the idea is to generate this `yaml` at compile-time using `godoc` (or the logic used by `godoc`)? I am familiar with the concept, but still intermediate with `golang`. Do you know if this is something supported by `godoc` directly or will require adaptation, or are you aware of a reference implementation used elsewhere?

Reply:
Yes, at compile-time that would be appropriate. I am not able to link to a specific implementation right now but the Swagger/OpenAPI generators do this -- maybe you could ask in #sig-docs?

Reply:
Okay thanks! I will do a bit more research on the matter, then reach out to the folks at #sig-docs  for guidance.

Reply:
The folks over at #sig-docs pointed me to a [ref impl](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/runtime/swagger_doc_generator.go) using go/ast module. I am working on a tool to extract this directly from the go source.

Reply:
@stevekuznetsov - it may take some time to generate the ast parser extract godoc Nodes and inject them into the `yaml`. In the meantime I am [generating `yaml` via `struct` marshaling](https://github.com/kubernetes/test-infra/pull/13507/files#diff-47374ed27f5081496337a98f327bd6bfR452) to make it *dynamic*.

Reply:
The YAML snippets are nice but without documentation on what the fields actually mean I would worry that providing the YAML will not help users, and they will still need to go read the struct comments :(

Reply:
@stevekuznetsov - I understand and agree. 👍 I will continue working on the yaml generator implementation as the acceptance criteria.

Reply:
@stevekuznetsov – I created a YAML generator #13637 inspired by (and evolved from) the swagger and kubectl generators in the `kubernetes/kubernetes` repo.